docs: add entry about pinning esm-infra and esm-apps (#2691)
canonical/ubuntu-pro-client: docs/explanations/about_esm.md

find out how, read our guide on
[enabling and disabling these services](../howtoguides/enable_esm_infra.md)
on your machine.

## Are `esm-infra` and `esm-apps` packages preferred over regular updates?

Yes. The Pro Client will deliver the following configuration files to apt to
ensure the `esm` updates are preferred when running `apt upgrade`:

- `/etc/apt/preferences.d/ubuntu-pro-esm-infra`
- `/etc/apt/preferences.d/ubuntu-pro-esm-apps`

Those files are *pinning* the respective `esm` origins to 510 - a value which

> is higher than the standard (500) defined in apt. Although the files will

Reviewer: Do we need to include the values? Because without already knowing what pinning is or how it works, the values could be really any values and hold no meaning for the reader. I think that if you want to include the values, we will need to explain what pinning is (however briefly). I'm also not 100% sure what you mean by "esm origins". It seems to imply there's more than one place to get esm updates from. What packages are supplied at the standard 500? (i.e., what is pinning esm to 510 giving preference over?)

Author: I removed specificities (as origins or values) and tried to better explain what is the impact about the pinning.
Please let me know if there is any more improvement needed.

find out how, read our guide on
[enabling and disabling these services](../howtoguides/enable_esm_infra.md)
on your machine.

## Are `esm-infra` and `esm-apps` packages preferred over regular updates?

Yes. The Pro Client will deliver the following configuration files to apt to
ensure the `esm` updates are preferred when running `apt upgrade`:

- `/etc/apt/preferences.d/ubuntu-pro-esm-infra`
- `/etc/apt/preferences.d/ubuntu-pro-esm-apps`

> Those files are *pinning* the packages coming from `esm` to a higher value than

Reviewer: ```suggestion
These files are *pinning* the packages coming from `esm` to give them a higher priority than
```

the standard defined in apt. This means that the version to be installed by apt
(in `install` or `upgrade`) or by unattended-upgrades will always be the

> highest esm version available for a given package, even if a higher version is

Reviewer: ```suggestion
highest `esm` version available for a given package, even if a higher version is
```

present on a non-esm source.

> This behavior guarantees that if you have esm-infra or esm-apps enabled, your

Reviewer: ```suggestion
This behavior guarantees that if you have `esm-infra` or `esm-apps` enabled, your
```

> system will always have the esm patches installed for any package available in

Reviewer: ```suggestion
system will always have the `esm` patches installed for any package available in
```

> the esm repositories.

Reviewer: ```suggestion
the ESM repositories.
```

Although the files will always be delivered by the package, they will only take
effect when the packages from those origins are available, i.e. when the

> services are enabled.

Reviewer: I found this a bit confusing because of the terminology switch. Which files are we referring to here? Do we mean that patches will be downloaded but not installed if the services are disabled?

Author: Well it means that, if the services are disabled, there is no harm in keeping those files around

Author: and the files are the preferences files... I tried to rewrite it.
